### DISCUSSION TITLE
Independant Checkpoint for hires.fix Plus Other Absent Features

### DISCUSSION THREAD

**Discussion by gohan2091** [vladmandic/sdnext#2610] [Q&A]
I'm using SD.Next and my online friend is using A1111. He is able to use a different model checkpoint for his hires.fix than from the first generation. So for example he can use RevAnimated then hires.fix with Realistic Vision all as one process. He showed me a setting to enable this. It appears (unless I've missed it) that this option is absent from SD.Next. Is that correct? Can this feature be added in please? 

There were other things he had in his A1111 that were absent. He has a sampler called "Restart". With SD XL checkpoints he was able to use a lot more samplers for example DPM ++ 3M SDE. His A1111 didn't require a restart of the server and backend change when switching between 1.5 and XL checkpoints. Also he had controlnet functioning for XL. 

Im quite happy to continue using SD.Next and I understand XL and controlnet XL are being worked on for better integration. I also heard the way A1111 got XL working is a messy crude way. Would love to try out the Restart sampler and be able to use a different model when I hires.fix though.

**Comment by vladmandic** [comment]
SDNext is not A1111 and settings are not the same. 
But yes, you CAN you different model for second pass - and you can even mix and match different model types for base and second pass.
And you definitely don't need to restart server to switch between SD15 and SDXL models. You need to restart server to switch between backends - which is not even close to same. Diffuser backend supports both and many other model types that can be loaded on the fly.

**Reply by gohan2091** [reply]
I do appreciate your reply and I am not saying A1111 and SD.Next are the same. I now feel tormented that you have confirmed I can run a different model for second pass but don't include in your reply how to do this. So I ask politely, how do I do this? Where is the setting? It's not in my UI, here is a screenshot of what I see:

![Screenshot 2023-12-08 205426](https://github.com/vladmandic/automatic/assets/127342824/a8a9e071-1110-4118-876d-e68d6dd5b9c6)

I've gone to the settings and searched for keywords like "second pass", "hires", "checkpoint" and I don't see anywhere to add this to the UI. In A1111, the setting tickbox is called "Hires fix: show hires checkpoint and sampler selection". 

Regarding 1.5 and XL. So I obviously am not understanding what a backend is, but anyway... so I just ran a test. I prompted "man" using a 1.5 checkpoint (epicphotogasm_lastUnicorn). Image generated fine. Then I switched checkpoint to an XL one (copaxTimelessxlSDXL1_v8) and prompted the same and I get this error in the UI:

```
Prompt:
Negative:
Parameters:
Error: model not loaded
Time: 5.86s
|
GPU active 50 MB reserved 50 | used 1708 MB free 22856 MB total 24564 MB | retries 0 oom 0
```
Here is the entire session text: https://pastebin.com/BKp6tnpy

So again, obviously I am missing something here. Changing between 1.5 models with the drop down box works fine: images generate. When switching to an XL model I always get that error. This is why I am confused by backend because the only way I can get XL models to generate images is switching the backend to diffusers mode and restarting the server. Please can you help me with these issues because I would very much like to continue to use SD.Next and I don't want to use A1111.

**Reply by vladmandic** [reply]
in backend=original, you can use sd15 and sd21 models. in backend=diffusers, you can use ALL models. diffusers is not just for sdxl.
regarding hires using different model, sdnext uses refiner for that and you can load refiner with any model you want. check wiki for details.

**Reply by gohan2091** [reply]
Thanks for replying. Ah so the independent checkpoint for hires is the diffuser drop down? Ok got it. I'll test that when I get home tomorrow. Thanks. However I'm still unclear on the xl models. The xl models only work on the diffusers backend switch for me. Setting the backend to original, the xl models don't work. So this is what I'm saying originally that I have to restart the server to switch between 1.5 and XL. You said I don't have to do that but in your most recent reply you said backend original is only for 1.5 and 2.1 models. So I'm a little confused. :(

**Reply by vladmandic** [reply]
No!
- backend:original is for SD15 and sd21
- backend:diffusers is for **ALL** model types, not just SDXL. For the third time!

**Reply by ETA-22** [reply]
The refiner dropdown is the second pass model when using diffusers backend for SD1.5/2./XL models.

**Reply by gohan2091** [reply]
Well you're writing the absolute bare minimum of replies and It's not really helping me to be honest. I am still stuck and haven't been able to switch between 1.5 and XL models on the fly. Are you saying to leave the backend on diffusers for good so I can use all types of models? if so, Controlnet then doesn't work, it gets disabled as well as several other plugins. So I may have XL working but then the 1.5 models have no ControlNet. I am really confused. If a member here could help me, I would be most appreciated.

@eadnams22 yes understood, I understood that part from Vladmandic but not the other part :(

**Reply by ETA-22** [reply]
Correct, no controlnet with diffusers.

**Reply by gohan2091** [reply]
Ok so then it seems to get XL working, I need the backend to be set to diffusers but by doing this I then lose access to controlnet even with 1.5 models. So basically if I've understood correctly, SD.Next is just not a good solution for XL use at the moment because I also want access to 1.5 things. Yet my friend who uses A1111 keeps asking me why I won't use A1111 as he has those features working himself. The ability to access 1.5 and XL on the fly and use controlnet.

**Reply by vladmandic** [reply]
each one has pro and cons and i'm not going to get into full list here.
post like "Yet my friend who uses A1111 keeps asking me why I won't use A1111" doesn't help. nobody is stopping you.

**Reply by ETA-22** [reply]
I don’t know why you would be losing any 1.5 model use with diffusers, I use 1.5 models all the time in diffusers.

**Comment by gohan2091** [comment]
> each one has pro and cons and i'm not going to get into full list here. post like "Yet my friend who uses A1111 keeps asking me why I won't use A1111" doesn't help. nobody is stopping you.

If I wanted to use A1111, I wouldn't be here. I am here because I like and support SD.Next.

> I don’t know why you would be losing any 1.5 model use with diffusers, I use 1.5 models all the time in diffusers.

No, that's not what I meant. I can use 1.5 models and XL models but by using diffusers backend, I lose the ability to have extensions like ControlNet so my 1.5 models don't have controlnet function unless I switch back to original backend but then I lose the XL models... that's what I am saying.

**Reply by vladmandic** [reply]
native controlnet for sdnext is coming soon.

**Comment by gohan2091** [comment]
Now with the new update, I was hoping my problem would be solved but nothing for me has changed. I am not sure if I am misunderstanding. To explain again... I want to use ControlNet with 1.5 models (and possibly XL models). I want both 1.5 and XL models to work on the fly without having to restart the server. If I go into diffusers backend, I can do this. If I go into standard backend, the 1.5 models work but the XL models do not (I get Error: model not loaded). When using diffuser backend, important extensions that I use all the time such as controlnet and multidiffusion gets disabled automatically.

Can someone please POLITELY help and explain?

**Reply by vladmandic** [reply]
diffusers backend now has controlnet support. 
multidiffusion is a no in that case, but i don't see what would that be a critical one. your choice.
